src/event_form/steps/event-form-step5.controller.spec.js

```diff
@@ -183,6 +265,515 @@ describe('Controller: event form step 5', function () {
+  });
+
+  it('should open the facilities modal', function () {
+    EventFormData.facilities = [];
+    spyOn(uibModal, 'open').and.returnValue({
+      result: $q.resolve()
+    });
+
+    scope.openFacilitiesModal();
+    scope.$apply();
+
+    expect(uibModal.open).toHaveBeenCalled();
+    expect(scope.facilitiesCssClass).toEqual('state-complete');
+    expect(scope.selectedFacilities).toEqual(EventFormData.facilities);
+    expect(scope.facilitiesInapplicable).toBeTruthy();
+  });
+
+  it('should fail in opening the facilities modal and set the state to incomplete', function () {
+    EventFormData.facilities = [];
+    spyOn(uibModal, 'open').and.returnValue({
+      result: $q.reject()
+    });
+
+    scope.openFacilitiesModal();
+    scope.$apply();
+
+    expect(uibModal.open).toHaveBeenCalled();
+    expect(scope.facilitiesCssClass).toEqual('state-incomplete');
+  });
+
+  it('should fail in opening the facilities modal and set the state to complete', function () {
+    EventFormData.facilities = ['faciliteit 1', 'faciliteit 2'];
+    spyOn(uibModal, 'open').and.returnValue({
+      result: $q.reject()
+    });
+
+    scope.openFacilitiesModal();
+    scope.$apply();
+
+    expect(uibModal.open).toHaveBeenCalled();
+    expect(scope.facilitiesCssClass).toEqual('state-complete');
+  });
+
+  it('should set the facilities on inapplicable', function () {
+    scope.setFacilitiesInapplicable();
+
+    expect(scope.facilitiesInapplicable).toBeTruthy();
+    expect(scope.facilitiesCssClass).toEqual('state-complete');
+  });
+
+  it('should remove all facilities and set it to inapplicable', function () {
+    EventFormData.facilities = ['faciliteit 1', 'faciliteit 2'];
+
+    eventCrud.updateFacilities.and.returnValue($q.resolve());
+
+    scope.setFacilitiesInapplicable();
+    scope.$apply();
+
+    expect(scope.facilitiesError).toBeFalsy();
+    expect(EventFormData.facilities).toEqual([]);
+
+    expect(scope.savingFacilities).toBeFalsy();
+    expect(scope.facilitiesInapplicable).toBeTruthy();
+    expect(scope.facilitiesCssClass).toEqual('state-complete');
+  });
+
+  it('should remove all facilities and handle the error', function () {
+    EventFormData.facilities = ['faciliteit 1', 'faciliteit 2'];
+
+    eventCrud.updateFacilities.and.returnValue($q.reject('pakot'));
+
+    scope.setFacilitiesInapplicable();
+    scope.$apply();
+
+    expect(EventFormData.facilities).toEqual([]);
+
+    expect(scope.savingFacilities).toBeFalsy();
+    expect(scope.facilitiesError).toBeTruthy();
+  });
+
+  it('should set the right toggle for the booking option "phone"', function () {
+    var contactInfos = [
+      {type: 'phone', value: '1234567890', booking: false},
+      {type: 'phone', value: '0987654321', booking: true}
+    ];
+    scope.bookingModel = {
+      phone: '0987654321',
+      url: 'http://google.be',
+      email: 'dude@sweet.com'
+    };
+
+    scope.showBookingOption(contactInfos[0]);
```

Comment:
@stijnswaanen `scope.showBookingOption(contactInfos[0])` is een keer op voorhand opgeroepen en opnieuw in de assertion iets lager. Geeft deze een ander resultaat wanneer meerdere keren opgeroepen? Als dit het geval is lijkt mij dat geen gewenst gedrag. Dit patroon herhaalt zich doorheen de tests...